### PR TITLE
[FIX] account: dashboard actions search filter 

### PR DESCRIPTION
**Pull request by agr-odoo**
Create a Bill
Register payment
Create the bank statement
Match with the payment
Go to accounting dashboard, hit "Vendor Bill" (open bill list view)
Clear filters
Add filter: Unpaid

Issue: Users should not be able to clear the journal filter when opening
the bill view

This commit align the behavior of the dashboard to the menu entries
by adding a default search domain to the actions instead of the default
search filter

opw-3228450

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by agr-odoo** [comment]
@oco-odoo

**Review by oco-odoo — CHANGES_REQUESTED** [review]
I'm not against the diff itself, but the scenario to reproduce the bug seems wrong. The "Bills" menu should only show vendor bills, not the payments/statements. So, in other words, the customer seems here to be complaining about the result of some custo, is he not ? The "Unpaid" filter and "payment state" fields are not displayed in the views showing non-invoice moves in standard.

**Comment by agr-odoo** [comment]
No customization are involved, the Unpaid filter is there and 'payment state' field is already visible by default
screenshot_049

**Comment by agr-odoo** [comment]
I just realized there at least two ways to open the same view:
1) (top menu) Vendors > Bills
2) Accounting Dashboard > Vendor Bills (card)

I can reproduce the issue only via 2)

**Comment by oco-odoo** [comment]
@agr-odoo Indeed, I had only checked with 1) .

I just talked about it with one of our POs (chkl), and we both agreed: it does not make sense to explicitly display the filter in the view that opens from 2) , because people will then be able to remove the filter, and will then view other moves than the one of the journal they clicked, including the ones for which the displayed fields (which corresponds to vendor bills in our example, but it also applies to customer invoices) make no sense at all. What do I care about the tax amount or bill date on a misc operation ? Why am I seeing my customer in the column "vendor" for customer invoices ?

Let's fix that by modifying this behavior: the filter should NOT be displayed;  the records should directly be filtered by the action, just like when using option 1).

File: addons/account/models/account_journal_dashboard.py [hunk 1]
```diff
@@ -677,19 +677,8 @@ def open_action(self):
...
-        domain_type_field = action['res_model'] == 'account.move.line' and 'move_id.move_type' or 'move_type' # The model can be either account.move or account.move.line
-
-        # Override the domain only if the action was not explicitly specified in order to keep the
-        # original action domain.
-        if not self._context.get('action_name'):
-            if self.type == 'sale':
-                action['domain'] = [(domain_type_field, 'in', ('out_invoice', 'out_refund', 'out_receipt'))]
-            elif self.type == 'purchase':
-                action['domain'] = [(domain_type_field, 'in', ('in_invoice', 'in_refund', 'in_receipt', 'entry'))]
-
```
Comment (oco-odoo) [inline addons/account/models/account_journal_dashboard.py:692]: How is this related ?

File: addons/account/models/account_journal_dashboard.py [hunk 2]
```diff
@@ -677,19 +677,8 @@ def open_action(self):
         action['context'] = context
         action['context'].update({
             'default_journal_id': self.id,
-            'search_default_journal_id': self.id,
```
Comment (oco-odoo) [inline addons/account/models/account_journal_dashboard.py:680]: That's right, but to me, you need to explicitly set action['domain'] = [('journal_id', '=', self.id)].

Without that, clicking misc journals on the dashboard opens all moves from other journals as well (invoice and bill journals seem to work differently ; you might want to check why it works for them ; we could be missing something important)

**Review by oco-odoo — CHANGES_REQUESTED** [review]

File: addons/account/models/account_journal_dashboard.py (hunk 2 above)
Comment (agr-odoo) [reply addons/account/models/account_journal_dashboard.py:680]: After double checking the behavior
When opening invoices we use `action_move_out_invoice_type` which has a domain defined 
`<field name="domain">[('move_type', '=', 'out_invoice')]</field>`
https://github.com/odoo/odoo/blob/c30c1dac33da05a4064d5353215098b2a5b3c87d/addons/account/views/account_move_views.xml#L1465
This is not the case when opening misc journals as we use `action_move_journal_line` with no domain
https://github.com/odoo/odoo/blob/c30c1dac33da05a4064d5353215098b2a5b3c87d/addons/account/views/account_move_views.xml#L1441

File: addons/account/models/account_journal_dashboard.py (hunk 1 above)
Comment (agr-odoo) [reply addons/account/models/account_journal_dashboard.py]: My bad, I didn't realize why this was necessary but I now get this is needed when opening the action via kanban card > 3 dots > (Invoices|CreditNote...)

**Review by oco-odoo — CHANGES_REQUESTED** [review]
Ok for me, but please adapt the commit message and PR description ;)

**Comment by agr-odoo** [comment]
@oco-odoo 
done :)

**Review by oco-odoo — APPROVED** [review]
@robodoo r+
